modules/consensus/synchronize.go

```diff
@@ -71,10 +110,35 @@ func blockHistory(tx *bolt.Tx) (blockIDs [32]types.BlockID) {
 }
 
 // threadedReceiveBlocks is the calling end of the SendBlocks RPC.
-func (cs *ConsensusSet) threadedReceiveBlocks(conn modules.PeerConn) error {
+func (cs *ConsensusSet) threadedReceiveBlocks(conn modules.PeerConn) (returnErr error) {
+	// Set a deadline after which SendBlocks will timeout. During IBD, esepcially,
+	// SendBlocks will timeout. This is by design so that IBD switches peers to
+	// prevent any one peer from stalling IBD.
+	err := conn.SetDeadline(time.Now().Add(sendBlocksTimeout))
+	// Ignore errors returned by SetDeadline if the conn is a pipe in testing.
+	// Pipes do not support Set{,Read,Write}Deadline and should only be used in
+	// testing.
+	if opErr, ok := err.(*net.OpError); ok && opErr.Op == "set" && opErr.Net == "pipe" && build.Release == "testing" {
+		err = nil
+	}
+	if err != nil {
+		return err
+	}
+	stalled := true
+	defer func() {
+		// KLUDGE: Timeout errors returned by muxado do not conform to the net.Error
```

Comment:
Can you use one of the existing keywords? TODO: is probably sufficient. If we have too many keywords, they become difficult to grep.